### PR TITLE
8.get /api/articles

### PR DESCRIPTION
**Pull request by CarageaAlexandru**
added functionality for GET /api/articles

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -121,6 +122,80 @@ describe("Endpoint integration", () => {
...
+							topic: expect.any(String),
+							created_at: expect.any(String),
+							votes: expect.any(Number),
+							comment_count: expect.any(Number),
+						});
+					});
+				});
+		});
+		test(`should accept the following query: topic which filters the topic by specified value
+			If the query is omitted the endpoing should respond with all articles`, () => {
```
Comment (MrMcSnail) [inline __tests__/integration.test.js:158]: spelling mistake/typo

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -121,6 +122,80 @@ describe("Endpoint integration", () => {
...
+				.get("/api/articles")
+				.expect(200)
+				.then(({ body }) => {
+					const { articles } = body;
+					expect(articles).toBeInstanceOf(Array);
+					expect(articles).toHaveLength(12);
+					expect(articles).toBeSortedBy("created_at", {
+						descending: true,
+					});
+					articles.forEach((article) => {
```
Comment (MrMcSnail) [inline __tests__/integration.test.js:145]: Nice length guarded forEach

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -197,4 +272,17 @@ describe("Error handing", () => {
...
 	});
+	describe("5. GET /api/articles", () => {
+		test(`should return status:200 if sorting query: "topic" does not exist in database.`, () => {
+			return request(app)
+				.get("/api/articles?topic=badInput")
+				.expect(200)
+				.then(({ body }) => {
+					const { message } = body;
+					expect(message).toBe(
+						"There are no matches based on specified topic in the database."
```
Comment (MrMcSnail) [inline __tests__/integration.test.js:283]: Nice error message

**Review by MrMcSnail — APPROVED** [review]
Merge! Merge my children!! MERGE!!

File: __tests__/integration.test.js (hunk 3 above)
Comment (CarageaAlexandru) [reply __tests__/integration.test.js:283]: thanks for the feedback!

**Pull request merged**